CONTRIBUTING.rst

```diff
@@ -115,19 +115,33 @@ For a more detailed discussion, read these :doc:`detailed documents
 
    * A pull request must be approved by two core team members before merging.
 
-5. Document changes
+.. _documenting-changes:
 
-   If your change introduces any API modifications, please update
-   ``doc/release/release_dev.rst``.
+5. Document changes
 
    If your change introduces a deprecation, add a reminder to ``TODO.txt``
    for the team to remove the deprecated functionality in the future.
 
+   If your change introduces any API modifications, or impacts users directly
+   (e.g. bug fixes, performance improvements) please describe the change inside
```

Comment:
This seems like overkill to me. I would prefer just saying that if you want to add a longer description of the PR you may.

Reply:
I agree with @jarrodmillman. First, API changes typically imply deprecation cycles, which brings us back to the previous sentence/instruction. Then, I'm afraid that whatever "impacts users directly (e.g. bug fixes, performance improvements) covers too broad an area.

Reply:
I included

> For trivial changes that are adequately summarized by the PR title, this
   block can be left empty (``...``).

below to make it clear that this is only required if the PR title isn't descriptive enough. Maybe I could have worded / ordered this better.

Basically, I want users to be able to read the New Features, API Changes, Bug fixes, and Performance sections without having to go to each PR to understand if this is relevant to them. E.g. consider https://github.com/scikit-image/scikit-image/pull/7166: out of context, the title

> Fix deprecation for phase_cross_correlation

is vague to users. But the extended summary makes it clear to users what's going on:

> Change number of returned arguments in `skimage.registration.phase_cross_correlation`. The function now always returns the 3 arguments `shift`, `error`, and `phasediff`.

I'll try to come with a wording that makes this clearer and has some tolerance build in.

Reply:
> below to make it clear that this is only required if the PR title isn't descriptive enough. Maybe I could have worded / ordered this better.

Ok, I got your intention right and I mostly agree. My reasoning is based on putting myself in a new contributor's shoes:

* repeating my support for @jarrodmillman's comment, I don't want to convey that something is (or isn't) *required*: We maintainers can complement whatever needs be complemented when we merge the PR;
* I would avoid calling any PR 'trivial:' In some cases it clearly is, by all standards, but in other, more subtle cases, it may sound demeaning.

Reply:
> is vague to users. But the extended summary makes it clear to users what's going on:

Sure, but I would simply *encourage* contributors to add a longer description of their PR...